### ISSUE TITLE
Would you be open to supporting promises?

### ISSUE THREAD

**Issue by alexbooker** [IonicaBizau/youtube-api#53]
Hi. Nice work on this module -- I'm using it for my site, [Community Casts](http://communitycasts.co/).  

I was wondering, would you be open to accepting a pull request that adds support for `Promise`s?

**Comment by IonicaBizau** [comment]
@alexbooker Definitely. But I guess the right place to implement it is directly in `googleapis` and then bump the dependency version here. /cc https://github.com/google/google-api-nodejs-client/issues/523

Also, you can promisify these callback-based functions anytime. :crystal_ball:

**Comment by IonicaBizau** [comment]
@alexbooker I upgraded the `googleapis` dependency in #56. Promise support was added in https://github.com/google/google-api-nodejs-client/commit/74fd9705f63621997359f26437ae0b3939c33feb so it just landed in this library too. :tada:

**Comment by alexbooker** [comment]
@IonicaBizau Awesome :+1:

**Comment by alexbooker** [comment]
Would you mind adding an example to the documentation?

I tried:

```javascript
const videoList = await youtubeApi.videos.list({
  id: videoId,
  part: 'snippet,contentDetails'
})
```

... But I'm not getting the result I was expecting. I'm getting a `Request` in return when I would expect a response.

**Comment by IonicaBizau** [comment]
@alexbooker I didn't test the code, but by reading the [code](https://github.com/google/google-api-nodejs-client/commit/74fd9705f63621997359f26437ae0b3939c33feb#diff-9475ffcd397ad130248cf92c9a813d65R204), I think you have to do something like this:


```js
const videoList = await youtubeApi.videos.list({
  id: videoId,
  part: 'snippet,contentDetails'
}).promise;
// ^ notice the promise field
```

**Comment by alexbooker** [comment]
@IonicaBizau Thanks, but I don't think that's it. 

![](http://i.imgur.com/ms1VtHE.png)

Notice the lack of **`await`**. I am inspecting the **`promise`** property and it's **`undefined`**:

![](http://i.imgur.com/7m2NBPf.png)

**Comment by alexbooker** [comment]
As far as I can tell, [the pull request you linked hasn't been merged :confused:](https://github.com/google/google-api-nodejs-client/pull/559)

**Comment by IonicaBizau** [comment]
@alexbooker Oh, that's correct... 😭

**Comment by alexbooker** [comment]
@IonicaBizau Maybe reopen this issue?

**Comment by IonicaBizau** [comment]
👍

**Comment by ulion** [comment]
@IonicaBizau when will the new version of this library be released? it's been 8 months since v2.1.7 and the promise feature should be released to public version asap?

**Comment by IonicaBizau** [comment]
@ulion Honestly, I don't have any idea, but maybe we can add the promise support in this library. We must make sure the last argument is the callback function always (in all the methods). Contributions are welcome in that direction!

For callback-compatibility, I would recommend using [`assured`](https://github.com/IonicaBizau/assured).
